types/tampermonkey/index.d.ts

```diff
@@ -531,3 +531,157 @@ declare function GM_setClipboard(
+        details: Tampermonkey.Request<TContext>, // tslint:disable-line:no-unnecessary-generics
+    ): Promise<Tampermonkey.Response<TContext>>;
+
+    // GM_download has two ways but GM.download doesn't
+    /**
+     * Downloads a given URL to the local disk
+     *
+     * @throws {Tampermonkey.DownloadErrorResponse}
+     */
+    download(details: Tampermonkey.DownloadRequest): Promise<void>;
+
+    // Tabs
+
+    /** Saves the tab object to reopen it after a page unload */
+    saveTab(obj: object): Promise<void>;
+
+    /** Gets a object that is persistent as long as this tab is open */
+    getTab(): Promise<any>;
+
+    /** Gets all tab objects as a hash to communicate with other script instances */
+    getTabs(): Promise<{ [tabId: number]: any }>;
+
+    // Utils
+    info: Tampermonkey.ScriptInfo;
+
+    /** Log a message to the console */
+    log(...message: any[]): Promise<void>;
+
+    /**
+     * Opens a new tab with this url.
+     * The options object can have the following properties:
+     * - `active` decides whether the new tab should be focused,
+     * - `insert` that inserts the new tab after the current one and
+     * - `setParent` makes the browser re-focus the current tab on close.
+     *
+     * Otherwise the new tab is just appended.
+     * If `options` is boolean (loadInBackground) it has the opposite meaning of
+     * active and was added to achieve Greasemonkey 3.x compatibility.
+     *
+     * If neither active nor loadInBackground is given, then the tab will not be
+     * focused.
+     * @returns Object with the function `close`, the listener `onclosed` and a flag
+     * called `closed`.
+     */
+    openInTab(url: string, options?: Tampermonkey.OpenTabOptions | boolean): Promise<Tampermonkey.OpenTabObject>;
+
+    /**
+     * Shows a HTML5 Desktop notification and/or highlight the current tab.
+     * @param ondone If specified used instead of `details.ondone`
+     * @returns True if the notification was clicked
+     */
+    notification(details: Tampermonkey.NotificationDetails, ondone?: Tampermonkey.NotificationOnDone): Promise<boolean>;
+
+    /**
+     * Shows a HTML5 Desktop notification and/or highlight the current tab.
+     * @param text Text of the notification
+     * @param title Notification title. If not specified the script name is used
+     * @param onclick Called in case the user clicks the notification
+     * @returns True if the notification was clicked
+     */
+    notification(
+        text: string,
+        title?: string,
+        image?: string,
+        onclick?: Tampermonkey.NotificationOnClick,
+    ): Promise<boolean>;
+
+    /**
+     * Copies data into the clipboard.
+     * The parameter 'info' can be an object like
+     * `{ type: 'text', mimetype: 'text/plain'}` or just a string expressing the
+     * type ("text" or "html").
+     */
+    setClipboard(
+        data: string,
+        info?: string | { type?: string | undefined; mimetype?: string | undefined },
```

Comment:
Please create type for `{ type?: string | undefined; mimetype?: string | undefined }`

Reply:
If I do that, should I also modify GM_setClipboard? Because I mostly just copied these from `GM_*`

Reply:
Sure.